File: tests/org/elixir_lang/reference/callable/Issue431Test.java
package org.elixir_lang.reference.callable;

import com.intellij.psi.PsiElement;
import com.intellij.testFramework.fixtures.BasePlatformTestCase;
import org.elixir_lang.psi.UnqualifiedNoArgumentsCall;

import static org.elixir_lang.reference.Callable.isParameter;

public class Issue431Test extends BasePlatformTestCase {
    /*
     * Tests
     */

    public void testIsParameter() {
        myFixture.configureByFiles("planet.ex");
        PsiElement parameter = myFixture
                .getFile()
                .findElementAt(myFixture.getCaretOffset())
                .getPrevSibling();

        assertInstanceOf(parameter, UnqualifiedNoArgumentsCall.class);
        assertTrue("planet is not marked as a parameter", isParameter(parameter));
    }

    /*
     * Protected Instance Methods
     */

    @Override
    protected String getTestDataPath() {
        return "testData/org/elixir_lang/reference/callable/issue_431";
    }
}

File: tests/org/elixir_lang/reference/callable/Issue431Test.kt
package org.elixir_lang.reference.callable

import org.elixir_lang.PlatformTestCase
import org.elixir_lang.psi.UnqualifiedNoArgumentsCall
import org.elixir_lang.reference.Callable.Companion.isParameter

class Issue431Test : PlatformTestCase() {
    fun testIsParameter() {
        myFixture.configureByFiles("planet.ex")
        val parameter = myFixture
                .file
                .findElementAt(myFixture.caretOffset)!!
                .prevSibling
        assertInstanceOf(parameter, UnqualifiedNoArgumentsCall::class.java)
        assertTrue("planet is not marked as a parameter", isParameter(parameter))
    }

    override fun getTestDataPath(): String = "testData/org/elixir_lang/reference/callable/issue_431"
}
